Andrey/trackers scrolling

andrey-p opened the pull request:
<!-- Please add the WIP label if the PR isn't complete. -->

**Reviewer:** @MariagraziaAlastra 

<!-- Optional fields
**CC:**
**Depends on:** 
-->

## Description:

This changes the tracker view to include the hero and explainer sections when scrolling, not just the tracker list. There was some weirdness with the container (see inline comments below) which is why the diff looks a bit crazier.

![screenshot from 2018-01-19 15-48-41](https://user-images.githubusercontent.com/1431577/35158939-8606c924-fd30-11e7-82c9-7d160a426d2e.png)

## Steps to test this PR:

Sorry, this change can potentially all the subviews in a tiny way.

1. Check the tracker view scrolls OK (e.g. go to https://vice.com and open the trackers tab)
2. The privacy practices should theoretically be able to scroll as well. I fudged it by adding `for (let i = 0; i < 10; i++) { good.push(good[0]) }` around [this line](https://github.com/duckduckgo/duckduckgo-privacy-extension/blob/develop/shared/js/ui/templates/privacy-practices.es6.js#L58) and visiting duckduckgo.com.
3. All other views/subviews should look exactly the same.


## Automated tests:
- [ ] Unit tests
- [ ] Integration tests

###### Reviewer Checklist:
- [ ] **Ensure the PR solves the problem**
- [ ] **Review every line of code**
- [ ] **Ensure the PR does no harm by testing the changes thoroughly**
- [ ] **Get help if you're uncomfortable with any of the above!**
- [ ] Determine if there are any quick wins that improve the implementation


###### PR Author Checklist:
- [ ] Get advice or leverage existing code
- [ ] Agree on technical approach with reviewer (if the changes are nuanced)
- [ ] Ensure that there is a testing strategy (and documented non-automated tests)
- [ ] Ensure there is a documented monitoring strategy (if necessary)
- [ ] Consider systems implications

MariagraziaAlastra commented:
Great work @andrey-p! Only thing: maybe it'd be cool to have a max width for the tracker name?
Low priority though, doesn't have to be in this PR
<img width="263" alt="screen shot 2018-01-19 at 1 59 41 pm" src="https://user-images.githubusercontent.com/3652195/35166778-0c4485d6-fd21-11e7-9a21-173299082d79.png">

andrey-p commented:
That's a good call, might as well fix it here!

andrey-p commented on shared/scss/popup.scss:
```diff
@@ -457,68 +461,6 @@ body {
-                            display: block;
-                            width: 65%;
-                            text-overflow: ellipsis;
-                            overflow: hidden;
```
Max width was already here. It's just none of those styles were applying because this was nested into `.site-info--details`, which wasn't happening in the new view.

I just copied the bits over that made sense to keep, and deleted the rest since it wasn't used.

MariagraziaAlastra replied:
@andrey-p I think the max-width and overflow logic needs a div to be effective:
<img width="2058" alt="screen shot 2018-01-19 at 2 27 59 pm" src="https://user-images.githubusercontent.com/3652195/35167875-fd3b1e98-fd24-11e7-9379-da6521efafdb.png">

vs

<img width="2023" alt="screen shot 2018-01-19 at 2 27 40 pm" src="https://user-images.githubusercontent.com/3652195/35167879-0363d56c-fd25-11e7-84cb-68ec82e5b155.png">

MariagraziaAlastra replied:
.url is currently a span, second screenshot is the current one

andrey-p replied:
Gah, sorry, I think I pushed without doublechecking. Lemme fix.

andrey-p commented:
@MariagraziaAlastra All right, fixed! I had to position the category absolutely, otherwise it was pushed down. I also added `white-space: nowrap` on the URL since it could wrap at a hyphen, and that looked weird.

MariagraziaAlastra commented:
Alright, LGTM! Thanks again @andrey-p!
